txscript/scriptnum.go

```diff
@@ -165,9 +160,9 @@ func (n scriptNum) Int32() int32 {
 // requireMinimal enabled.
 //
 // See the Bytes function documentation for example encodings.
-func makeScriptNum(v []byte, requireMinimal bool) (scriptNum, error) {
-	// Interpreting data as an integer requires that it is not larger than
-	// a 32-bit integer.
+func makeScriptNum(v []byte, requireMinimal bool, maxScriptNumLen int) (scriptNum, error) {
+	// Interpreting data requires that it is not larger than
```

Comment:
the "as an integer" part should still be retained.  Regardless of the number of bytes, it's still being interpreted as an integer.